### PR TITLE
fix the issue that @property types are not recoganized

### LINKED ISSUE #9620: Object member type defined by JSDoc @property tags are not recoganized
From: https://github.com/Microsoft/TypeScript/issues/9276#issuecomment-231597333

https://github.com/Microsoft/TypeScript/pull/8103 
Format 3 doesn't work properly

```javascript
/**
 * @typedef {Object} Product
 * @property {String} name
 */

/**
 * Creates a product instance
 * @return {Product}
 */
function makeProduct() {}
```
screen shot 2016-07-10 at 9 14 20 pm

-------------
```
$ tsc -v
Version 2.0.0-dev.20160707
```

### PR DESCRIPTION
**Pull request by zhengbli**
Fixes #9620

### CODE REVIEW

File: src/compiler/checker.ts [hunk 1]
```diff
@@ -3160,6 +3160,9 @@ namespace ts {
                 if (declaration.kind === SyntaxKind.ExportAssignment) {
                     return links.type = checkExpression((<ExportAssignment>declaration).expression);
                 }
+                if (declaration.kind === SyntaxKind.JSDocPropertyTag && (<JSDocPropertyTag>declaration).typeExpression) {
```
Comment (mhegazy) [inline src/compiler/checker.ts]: should not this be in getTypeForVariableLikeDeclarationFromJSDocComment

Comment (zhengbli) [inline src/compiler/checker.ts]: That method is originally for declarations written in the code whose type annotation in another seperated jsdoc comment, so other methods who call it all assumes `VariableLikeDeclaration` type of the declaration. It seems to be a lot of change to make all of them accept `VariableLikeDeclaration | JSDocPropertyTag`. What would be the benefit of doing that?

Comment (mhegazy) [inline src/compiler/checker.ts]: nothing specifically. it is just where all our get-type-from-jsdoc code.

**Comment by mhegazy** [comment]
👍

**Pull request merged**